test/functional/helpers/createFixture/clientScripts.js

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import path from "path";
+import fs from "fs";
+import { ClientFunction } from "testcafe";
+
+const edgeEnv = process.env.EDGE_ENV || "int";
+const alloyEnv = process.env.ALLOY_ENV || "int";
+
+// eslint-disable-next-line no-console
+console.log("EDGE ENV:", edgeEnv);
+// eslint-disable-next-line no-console
+console.log("ALLOY ENV:", alloyEnv);
+
+const localPromisePolyfillPath = path.join(
+  __dirname,
+  "../promisePolyfill/promise-polyfill.min.js"
+);
+const remotePromisePolyfillPath =
+  "https://cdn.jsdelivr.net/npm/promise-polyfill@8/dist/polyfill.min.js";
+const remoteVisitorLibraryUrl =
+  "https://github.com/Adobe-Marketing-Cloud/id-service/releases/latest/download/visitorapi.min.js";
+const alloyPageSnippetPath = path.join(
+  __dirname,
+  "../alloyPageSnippet/index.js"
+);
+const localAlloyLibraryPath = path.join(
+  __dirname,
+  "../../../../dist/standalone/alloy.js"
+);
+const remoteAlloyLibraryUrl =
+  "https://cdn1.adoberesources.net/alloy/latest/alloy.js";
+
+let localAlloyCode;
+
+// We're only testing against the alloy code found in /dist when alloyEnv
+// is "int". Otherwise, we're testing against the prod alloy that's on the CDN
+// and in those cases alloy code may not event exist in /dist. This is why
+// we only try to load the file from the file system if alloyEnv is int.
+if (alloyEnv === "int") {
+  localAlloyCode = fs.readFileSync(localAlloyLibraryPath, "utf8");
+}
+
+const addRemoteUrlClientScript = ({ clientScripts, url, async = false }) => {
+  // TestCafe client scripts don't "natively" support loading a script
+  // from a remote URL, so we have to make our local script add another script
+  // element that loads Visitor from the remote server.
```

Comment:
Note to self: I mention "Visitor" here because I wrote the comment before I made this function generic for any remote URL. I should remove "Visitor" from the comment.